src/main/java/org/radarcns/management/repository/ProjectRepository.java

```diff
@@ -1,35 +1,42 @@
 package org.radarcns.management.repository;
 
+import java.util.List;
 import java.util.Optional;
-import org.radarcns.management.domain.SourceType;
 import org.radarcns.management.domain.Project;
+import org.radarcns.management.domain.SourceType;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 
-import java.util.List;
-
 /**
  * Spring Data JPA repository for the Project entity.
  */
 @SuppressWarnings("unused")
-public interface ProjectRepository extends JpaRepository<Project,Long> {
+public interface ProjectRepository extends JpaRepository<Project, Long> {
 
-    @Query("select distinct project from Project project left join fetch project.sourceTypes")
-    List<Project> findAllWithEagerRelationships();
+    @Query(value = "select distinct project from Project project "
+            + "left join fetch project.sourceTypes",
+            countQuery = "select distinct count(project) from Project project")
+    Page<Project> findAllWithEagerRelationships(Pageable pageable);
 
-    @Query("select project from Project project left join fetch project.sourceTypes where project.id =:id")
+    @Query("select project from Project project left join fetch "
+            + "project.sourceTypes where project.id =:id")
     Project findOneWithEagerRelationships(@Param("id") Long id);
 
-    @Query("select project from Project project left join fetch project.sourceTypes where project.projectName =:name")
+    @Query("select project from Project project left join fetch "
+            + "project.sourceTypes where project.projectName =:name")
     Project findOneWithEagerRelationshipsByName(@Param("name") String name);
 
     @Query("select project.sourceTypes from Project project WHERE project.id = :id")
     List<SourceType> findSourceTypesByProjectId(@Param("id") Long id);
 
-    @Query("select distinct sourceType from Project project left join project.sourceTypes sourceType "
-        + "where project.id =:id "
-        + "and sourceType.id = :sourceTypeId ")
+    @Query(
+            "select distinct sourceType from Project project "
```

Comment:
Could probably be on the previous line: `@Query("select distinct ...` just as the other multi-line queries